client/src/components/TripElement/EditTripForm.js

```diff
@@ -1,27 +1,26 @@
 import { useDispatch, useSelector } from 'react-redux'
 import { closeEditTripModal } from '../../redux/reducers/modalsSlice'
-import { selectTrips, editTrip } from '../../redux/reducers/userSlice'
+import { editTripAsync } from '../../redux/reducers/trip/thunks'
 import { Modal } from '../common'
 import { TripForm } from './TripForm'
 
 export function EditTripForm() {
+  const dispatch = useDispatch()
+  const trips = useSelector((state) => state.trip.trips)
+
   const activeTripId = useSelector((state) => state.view.activeTripId)
-  const trips = useSelector(selectTrips)
-  const activeTrip = trips.find((trip) => trip.id === activeTripId)
+  const activeTrip = trips.find((trip) => trip._id === activeTripId)
 
   const editTripModalIsOpen = useSelector(
     (state) => state.modals.editTripModalIsOpen,
   )
-  const dispatch = useDispatch()
 
   const handleCloseEditTripModal = () => {
     dispatch(closeEditTripModal())
   }
 
   const onSubmit = (data) => {
-    // TODO: make api call to create new trip
-    console.log(data)
-    dispatch(editTrip({ ...activeTrip, ...data }))
+    dispatch(editTripAsync({ id: data._id, tripData: data }))
```

Comment:
nit: if there isnt a situation where the `id` param is different from the one in `tripData` we should send the object directly.

Reply:
> we would have to separate them when doing the PATCH/PUT call anyways.

why the seperation for PATCH/PUT? 

> we can either do it here or in the server.

Did you have it like this for readability?

Reply:
typically PUT and PATCH would need id in URL and the changes in the body.

Reply:
ok lgtm then